site/content/entry/improvements-to-hardware-renderer.md

```diff
@@ -0,0 +1,200 @@
+In a momentous occasion, Citra displayed 3D graphics from a commercial game, Legend of Zelda: Ocarina of Time 3D
+
+{{< figure type="youtube" id="sYukdJam6gk" title="It took a year from when citra started to get to this point" >}}
+
+This engineering feat was thanks to the hardwork of many contibutors in both the emulator scene and the 3ds hacking scene, who worked tirelessly to reverse engineer the 3DS GPU, a chip called the PICA200.
+But not even a few months later, Citra was able to play the game at fullspeed! 
+
+{{< figure type="youtube" id="Hj8sPsB5qXQ" title="Good thing that shield flashing bug still isn't around" >}}
+
+Why is there such a major difference in speed from the first and the second video?
+The speed difference boils down to how the 3DS GPU is being emulated.
+The first video is showing off the software renderer, which emulates the PICA200 by using your computer's CPU.
+On the other hand, the second video is using the OpenGL hardware renderer, which emulates the PICA200 by using your computer's GPU.
+From those videos, using your GPU to emulate the 3DS GPU is the clear winner when it comes to speed!
+However, it's not all sunshine and daisies; there's always tradeoffs in software development.
+
+## Challenges in the Hardware Renderer
+
+Earlier it was stated that the OpenGL hardware renderer was emulating the PICA200 by using the GPU instead of the CPU, and ... that's only partially true.
+As it stands, only a portion of the PICA200 emulation is running on the GPU; most of it is running on the CPU!
+To understand why, we need to dive a little deeper into the difference between CPUs and modern GPUs.
+
+{{< figure src="/images/entry/improvements-to-hardware-renderer/gpu_pipeline_before.png" 
+    title="The green shaded portions are what Citra emulates using the GPU. For a hardware renderer, Citra isn't using the GPU much!" 
+>}}
+
+As a general rule of thumb, CPUs are fast at computing general tasks, while GPUs are *blazing fast* at computing very specific tasks.
+Whenever the tasks the PICA200 performs matches well with tasks you can do on a GPU using OpenGL, everything is fast and everyone is happy!
+That said, we tend to run into edge cases that the PICA200 supports, but frankly, OpenGL is not well suited to support.
+This leads to cases where sometimes we just have to [live with minor inaccuracies as a tradeoff for speed](https://github.com/citra-emu/citra/pull/2697)
+
+{{< figure src="/images/entry/improvements-to-hardware-renderer/outline_bug.png" title="The infamous Pokémon outlines bug was one such example of something that OpenGL just doesn't handle well" >}}
+
+OpenGL is also great for emulator developers because it's a cross-platform standard for graphics, with support for all major desktop platforms.
+But because it's a standard, this means performance and features can vary widely between operating systems, graphics driver, and the physical graphics card.
```

Comment:
Because it's a standard is the reason performance and features vary? Can we give better reasons than that?